substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrManager.java

```diff
@@ -0,0 +1,255 @@
+/*
+ * Copyright (c) 2019, 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.jfr;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.graalvm.compiler.api.replacements.Fold;
+import org.graalvm.compiler.debug.TTY;
+import org.graalvm.compiler.options.Option;
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.Platform;
+import org.graalvm.nativeimage.Platforms;
+
+import com.oracle.svm.core.option.RuntimeOptionKey;
+import com.oracle.svm.core.util.UserError;
+import com.oracle.svm.core.util.UserError.UserException;
+import com.oracle.svm.jfr.events.ClassLoadingStatistics;
+import com.oracle.svm.jfr.events.InitialEnvironmentVariable;
+import com.oracle.svm.jfr.events.InitialSystemProperty;
+import com.oracle.svm.jfr.events.JVMInformation;
+import com.oracle.svm.jfr.events.JavaThreadStatistics;
+import com.oracle.svm.jfr.events.OSInformation;
+import com.oracle.svm.jfr.events.PhysicalMemory;
+
+import jdk.jfr.FlightRecorder;
+
+/**
+ * Called during VM startup and teardown. Also triggers the JFR argument parsing.
+ */
+public class JfrManager {
+    private static final String DEFAULT_JFC_NAME = "default";
+
+    @Platforms(Platform.HOSTED_ONLY.class)
+    public JfrManager() {
+    }
+
+    @Fold
+    static JfrManager get() {
+        return ImageSingletons.lookup(JfrManager.class);
+    }
+
+    void setup() {
+        if (Options.FlightRecorder.getValue()) {
+            periodicEventSetup();
+            initRecording();
+        }
+    }
+
+    void teardown() {
+        if (Options.FlightRecorder.getValue()) {
+            // Everything should already have been torn down by JVM.destroyJFR(), which is called in
+            // a shutdown hook.
+            assert !SubstrateJVM.isInitialized();
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Use of identifier that is a keyword in later versions of Java (NM_FUTURE_KEYWORD_USED_AS_IDENTIFIER)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-use-of-identifier-that-is-a-keyword-in-later-versions-of-java-nm-future-keyword-used-as-identifier).
The identifier is a word that is reserved as a keyword in later versions of Java, and your code will need to be changed in order to compile it in later versions of Java.